From 1f6b213d92e71de1eb669e9fb869d564066dadf5 Mon Sep 17 00:00:00 2001
From: Vishvananda Ishaya <vishvananda@gmail.com>
Date: Mon, 26 Sep 2011 08:34:14 -0700
Subject: [PATCH 09/40] Fix the grantee group loading for source groups

Fixes bug 859679.

(cherry picked from commit 114ca0b2dc090270d8d20bd10b6e1e7a0b451db2)

Change-Id: I92569ecc50fe3e9f17f633df03ee428587749a48
---
 nova/db/sqlalchemy/api.py |    5 ++---
 1 files changed, 2 insertions(+), 3 deletions(-)

diff --git a/nova/db/sqlalchemy/api.py b/nova/db/sqlalchemy/api.py
index 3226999..6cc678a 100644
--- a/nova/db/sqlalchemy/api.py
+++ b/nova/db/sqlalchemy/api.py
@@ -2828,14 +2828,13 @@ def security_group_rule_get_by_security_group(context, security_group_id,
         result = session.query(models.SecurityGroupIngressRule).\
                          filter_by(deleted=can_read_deleted(context)).\
                          filter_by(parent_group_id=security_group_id).\
-                         options(joinedload_all('grantee_group')).\
+                         options(joinedload_all('grantee_group.instances')).\
                          all()
     else:
-        # TODO(vish): Join to group and check for project_id
         result = session.query(models.SecurityGroupIngressRule).\
                          filter_by(deleted=False).\
                          filter_by(parent_group_id=security_group_id).\
-                         options(joinedload_all('grantee_group')).\
+                         options(joinedload_all('grantee_group.instances')).\
                          all()
     return result
 
-- 
1.7.6.4

